yarn-project/simulator/src/public/avm/avm_machine_state.ts

```diff
@@ -38,6 +38,8 @@ export class AvmMachineState {
   public pc: number = 0;
   /** program counter of the next instruction, byte based */
   public nextPc: number = 0;
+  /** instruction counter, including nested calls */
+  public instrCounter: number = 0;
```

Comment:
Should we move this to a separate section of "metrics only, not needed for execution"?